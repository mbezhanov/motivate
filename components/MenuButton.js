import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import { AntDesign, Octicons } from '@expo/vector-icons';
import StyledText from './StyledText';
import Sizes from "../constants/Sizes";
import Colors from '../constants/Colors';

const button = (props) => {
  const Icon = ['upload', 'download'].indexOf(props.icon) !== -1 ? AntDesign : Octicons;

  return (
    <View style={styles.container}>
      <View style={styles.textContainer}>
        <StyledText style={{ marginHorizontal: Sizes.S, marginVertical: Sizes.S / 2 }}>{props.label}</StyledText>
      </View>
      <TouchableOpacity style={styles.button} onPress={props.onPress}>
        <Icon name={props.icon} size={Sizes.M} color={Colors.HIGHLIGHT} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    marginBottom: Sizes.L / 2,
    flexDirection: 'row',
  },
  button: {
    width: Sizes.L * 2,
    height: Sizes.L * 2,
    borderRadius: Sizes.L,
    textAlign: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.BACKGROUND,
  },
  textContainer: {
    textAlign: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.BACKGROUND_TRANSPARENT,
    borderRadius: Sizes.S / 2,
    marginRight: Sizes.S / 2,
  },
});

export default button;
