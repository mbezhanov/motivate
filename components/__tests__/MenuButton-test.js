import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import MenuButton from '../MenuButton';

jest.mock('TouchableOpacity', () => 'TouchableOpacity');
jest.mock('../StyledText', () => 'StyledText');
jest.mock('@expo/vector-icons', () => {
  const React = require('React');
  class AntDesign extends React.Component {
    render() {
      return React.createElement('AntDesign', this.props, this.props.children)
    }
  }
  class Octicons extends React.Component {
    render() {
      return React.createElement('Octicons', this.props, this.props.children)
    }
  }
  return { AntDesign, Octicons };
});

describe('MenuButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MenuButton label="foo" icon="bar" onPress={() => {}} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('uses a different icon set for the "upload" and "download" icons', () => {
    const a = renderer.create(<MenuButton label="foo" icon="bar" onPress={() => {}} />).toJSON();
    const b = renderer.create(<MenuButton label="foo" icon="upload" onPress={() => {}} />).toJSON();
    const c = renderer.create(<MenuButton label="foo" icon="download" onPress={() => {}} />).toJSON();
    let diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
    diff = snapshotDiff(a, c);
    expect(diff).toMatchSnapshot();
  });
});
