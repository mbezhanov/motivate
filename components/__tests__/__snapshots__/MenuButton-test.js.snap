// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginBottom": 10.5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(43, 43, 43, 0.5)",
        "borderRadius": 6,
        "justifyContent": "center",
        "marginRight": 6,
        "textAlign": "center",
      }
    }
  >
    <StyledText
      style={
        Object {
          "marginHorizontal": 12,
          "marginVertical": 6,
        }
      }
    >
      foo
    </StyledText>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2b2b2b",
        "borderRadius": 21,
        "height": 42,
        "justifyContent": "center",
        "textAlign": "center",
        "width": 42,
      }
    }
  >
    <Octicons
      color="#fff"
      name="bar"
      size={16}
    />
  </TouchableOpacity>
</View>
`;

exports[`MenuButton uses a different icon set for the "upload" and "download" icons 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -43,12 +43,12 @@
          \\"textAlign\\": \\"center\\",
          \\"width\\": 42,
        }
      }
    >
-     <Octicons
+     <AntDesign
        color=\\"#fff\\"
-       name=\\"bar\\"
+       name=\\"upload\\"
        size={16}
      />
    </TouchableOpacity>
  </View>"
`;

exports[`MenuButton uses a different icon set for the "upload" and "download" icons 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -43,12 +43,12 @@
          \\"textAlign\\": \\"center\\",
          \\"width\\": 42,
        }
      }
    >
-     <Octicons
+     <AntDesign
        color=\\"#fff\\"
-       name=\\"bar\\"
+       name=\\"download\\"
        size={16}
      />
    </TouchableOpacity>
  </View>"
`;
